lib/newspaper_works/ingest/lc_publication_info.rb

```diff
@@ -0,0 +1,146 @@
+require 'faraday'
+require 'nokogiri'
+require 'uri'
+
+module NewspaperWorks
+  module Ingest
+    # rubocop:disable Metrics/ClassLength
+    class LCPublicationInfo < BasePublicationInfo
+      attr_accessor :place_of_publication, :full_title, :lccn, :place_name, :doc
+
+      XML_NS = {
+        mods: 'http://www.loc.gov/mods/v3',
+        MODS: 'http://www.loc.gov/mods/v3'
+      }.freeze
+
+      BASE_URL = 'https://lccn.loc.gov'.freeze
+
+      def initialize(lccn)
+        super(lccn)
+        @doc = nil
+        @full_title = nil
+        @place_of_publication = nil
+        @place_name = nil
+        load
+      end
+
+      def inspect
+        format(
+          "<#{self.class}:0x000000000%<oid>x " \
+            "\tlccn: '#{@lccn}'>",
+          oid: object_id << 1
+        )
+      end
+
+      def url
+        "#{BASE_URL}/#{@lccn}/mods"
+      end
+
+      def load_lc
+        resp = Faraday.get url
+        @doc = Nokogiri.XML(resp.body)
+        return if empty?
+        # try title[@type="uniform"] first:
+        title = find('//mods:titleInfo[@type="uniform"]/mods:title').first
+        # if no type="uniform" title, try non-alternate bare titleInfo:
+        #   -- in either case, should omit any non-sorted article (e.g. "The")
+        title = find('//mods:titleInfo[count(@type)=0]/mods:title').first if title.nil?
+        @full_title = title.text unless title.nil?
+      end
+
+      def mods_place_name
+        # prefer geographic subject hierarchy for place name construction:
+        city = find('//mods:hierarchicalGeographic/mods:city').first
+        # State (e.g. "Utah"), Province (e.g. "Ontario"), other (e.g. "England")
+        state = find('//mods:hierarchicalGeographic/mods:state').first
+        # if state is nil, fallback to country in its place
+        state = find('//mods:hierarchicalGeographic/mods:country').first if state.nil?
+        return "#{city.text}, #{state.text}" if city && state
+        # fallback to placeTerm text, which may be abbreviated in such a
+        #   way that geonames struggles to find on search; for a list of
+        #   abbreviations, see:
+        #   https://www.loc.gov/aba/publications/FreeSHM/H0810.pdf
+        name = find('//mods:originInfo//mods:placeTerm[@type="text"]').first
+        name.nil? ? nil : name.text
+      end
+
+      def load_place
```

Comment:
Same comment as with `ChronAmPublicationInfo#load_place`. The value of `@place_of_publication` needs to be `Hyrax::ControlledVocabularies::Location.new(uri)`.